themes/default/content/blog/troubleshooting-builds-with-diagnostics/index.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Troubleshooting Builds With Diagnostics"
+date: 2022-03-29T12:00:55-06:00
+draft: false
+meta_desc: Explore how to gather diagnostic data to debug your Pulumi programs.
+meta_image: meta.png
+authors:
+    - laura-santamaria
+tags:
+    - troubleshooting
+---
+
+One of the most common things a developer, an operations professional, a security researcher, and a quality assurance engineer all have in common is the pain of troubleshooting. No matter where you are in an IT organization, you probably have cursed your system at some point as you stare at a wall of characters representing an error somewhere. Whenever humans interact with a system, we introduce chaos, and the more complex the system, the more likely the system's diagnostic data is complex and multi-layered. Pulumi, as a system that represents some of the more complex systems we can make, is no exception. So let's demystify diagnostics in Pulumi.
+
+<!--more-->
+
+## Getting data
+
+First, if you don't know how to get diagnostic data from a Pulumi run, our [troubleshooting page]({{< relref "/docs/troubleshooting" >}}) is a good place to start. There's a section specifically on [diagnostics]({{< relref "/docs/troubleshooting#diagnosing-issues" >}}). However, let's dig in on this a bit.
+
+### Data sources
+
+There's two different kinds of diagnostics you can get from a standard Pulumi CLI command because there's actually two pieces running: the Pulumi [deployment engine]({{< relref "/docs/intro/concepts/how-pulumi-works#deployment-engine" >}}) and the [providers]({{< relref "/docs/reference/glossary#resource-provider" >}}) (resource providers, cloud providers, etc.). To get data from each piece, you need to run different flags. In general, the root flag you'll always need is the `-v` flag, which stands for "verbose" (that's standard for most CLIs). That flag gives you data from the deployment engine. So, if you want general diagnostics for a `pulumi up`, you would add the `-v` flag with a number representing the logging level you need:
```

Comment:
> resource providers, cloud providers, etc.

I'd just call this "resource providers" and avoid extra terms that aren't part of our lexicon.

Reply:
Good call. Done.